boltdesignsystem/bolt
# DS-236: Add support for images in action blocks (and other cleanup)

## Jira

https://pegadigitalit.atlassian.net/browse/DS-236

## Summary

Allows images and other freeform content to be passed as content in action block items

## Details

- Removes the `icon` property (with deprecation) and adds a new `content` property.   There was an unused `__icon` class that was also removed.  Now, instead of passing an object for the icon, you can pass an already rendered icon or anything else for content.
- Renames `contentItems` to `items` (with deprecation)
- Adds support for passing an item without a url.  This makes something like the partner links shown at https://boltdesignsystem.com/pattern-lab/?p=pages-event-detail possible.
- Updates the partner links to require less custom CSS and delegate more to action blocks.
- Replaces the private _action-blocks-item.twig file with a public action-block.twig file.  This was based on a discussion with @danielamorse about architecture for patterns with multiple items.

## How to test

- Confirm that images and other content can now be passed to action blocks as shown in the new example at /pattern-lab/?p=components-action-blocks-freeform-content
- Confirm that action blocks examples shown at the following URLs are unchanged.
/pattern-lab/patterns/40-components-action-blocks/index.html
/pattern-lab/?p=pages-event-detail
